static/components.ts

```diff
@@ -22,100 +22,54 @@
     CONFORMANCE_VIEW_COMPONENT_NAME,
-    CompilerForTreeState,
     ComponentConfig,
+    ComponentStateMap,
     DEVICE_VIEW_COMPONENT_NAME,
     DIFF_VIEW_COMPONENT_NAME,
+    DragSourceFactory,
     EDITOR_COMPONENT_NAME,
     EXECUTOR_COMPONENT_NAME,
-    EmptyAstViewState,
-    EmptyCfgViewState,
-    EmptyClangirViewState,
-    EmptyCompilerState,
-    EmptyDeviceViewState,
-    EmptyDiffViewState,
-    EmptyEditorState,
-    EmptyExecutorState,
-    EmptyFlagsViewState,
-    EmptyGccDumpViewState,
-    EmptyGnatDebugTreeViewState,
-    EmptyGnatDebugViewState,
-    EmptyHaskellCmmViewState,
-    EmptyHaskellCoreViewState,
-    EmptyHaskellStgViewState,
-    EmptyIrViewState,
-    EmptyOptPipelineViewState,
-    EmptyOptViewState,
-    EmptyPpViewState,
-    EmptyRustHirViewState,
-    EmptyRustMacroExpViewState,
-    EmptyRustMirViewState,
-    EmptyStackUsageViewState,
-    EmptyToolInputViewState,
-    EmptyTreeState,
-    ExecutorForTreeState,
     FLAGS_VIEW_COMPONENT_NAME,
     GCC_DUMP_VIEW_COMPONENT_NAME,
     GNAT_DEBUG_TREE_VIEW_COMPONENT_NAME,
     GNAT_DEBUG_VIEW_COMPONENT_NAME,
+    GoldenLayoutConfig,
     HASKELL_CMM_VIEW_COMPONENT_NAME,
     HASKELL_CORE_VIEW_COMPONENT_NAME,
     HASKELL_STG_VIEW_COMPONENT_NAME,
     IR_VIEW_COMPONENT_NAME,
+    ItemConfig,
+    LLVM_OPT_PIPELINE_VIEW_COMPONENT_NAME,
+    LayoutItem,
     OPT_PIPELINE_VIEW_COMPONENT_NAME,
     OPT_VIEW_COMPONENT_NAME,
     OUTPUT_COMPONENT_NAME,
-    OutputState,
     PP_VIEW_COMPONENT_NAME,
-    PopulatedAstViewState,
-    PopulatedCfgViewState,
-    PopulatedClangirViewState,
-    PopulatedCompilerState,
-    PopulatedConformanceViewState,
-    PopulatedDeviceViewState,
-    PopulatedDiffViewState,
-    PopulatedEditorState,
-    PopulatedExecutorState,
-    PopulatedFlagsViewState,
-    PopulatedGccDumpViewState,
-    PopulatedGnatDebugTreeViewState,
-    PopulatedGnatDebugViewState,
-    PopulatedHaskellCmmViewState,
-    PopulatedHaskellCoreViewState,
-    PopulatedHaskellStgViewState,
-    PopulatedIrViewState,
-    PopulatedOptPipelineViewState,
-    PopulatedOptViewState,
-    PopulatedPpViewState,
-    PopulatedRustHirViewState,
-    PopulatedRustMacroExpViewState,
-    PopulatedRustMirViewState,
-    PopulatedStackUsageViewState,
-    PopulatedToolInputViewState,
     RUST_HIR_VIEW_COMPONENT_NAME,
     RUST_MACRO_EXP_VIEW_COMPONENT_NAME,
     RUST_MIR_VIEW_COMPONENT_NAME,
     STACK_USAGE_VIEW_COMPONENT_NAME,
     TOOL_COMPONENT_NAME,
     TOOL_INPUT_VIEW_COMPONENT_NAME,
     TREE_COMPONENT_NAME,
-    ToolViewState,
 } from './components.interfaces.js';
-import {ConfiguredRuntimeTools} from './execution/execution.interfaces.js';
-import {LanguageKey} from './languages.interfaces.js';
 
 /** Get an empty compiler component. */
-export function getCompiler(editorId: number, lang: string): ComponentConfig<EmptyCompilerState> {
+export function getCompiler(editorId: number, lang: string): ComponentConfig<typeof COMPILER_COMPONENT_NAME> {
```

Comment:
Maybe it's preferable to create aliases for all the component configs? I think it's a lot more clear if we have a `CompilerComponentConfig` instead of the typeof NAME.

```ts
type CompilerComponentConfig = ComponentConfig<typeof COMPILER_COMPONENT_NAME>
```

Reply:
We did consider this but it leads to duplication

Reply:
>>> While the type aliases are more readable, they come with a significant cost:

  1. Maintenance burden: We'd need to add a type alias for every component type (20+ aliases)
  2. Import complexity: Files would need to import both the constant and the type alias
  3. Duplication: We're creating another name that needs to be kept in sync

  The current typeof pattern actually has these benefits:
  1. Zero maintenance: No additional aliases to maintain
  2. Single source of truth: The constant name drives everything
  3. Type safety: If you misspell the constant, TypeScript catches it immediately